src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,471 @@
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  pointerEvents({ disabled }) {
+    if (disabled) {
+      return 'none';
+    }
+    return '';
+  },
+};
+
+const StyledButton = styled.button`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${(props) => props.theme.spacings.xxsmall};
+  border: ${styles.border};
+  width: ${(props) => (props.block ? '100%' : '')};
+  cursor: pointer;
+  pointer-events: ${styles.pointerEvents};
+  outline: none;
+  &:hover {
+    border: ${styles.hoverBorder};
+    background-color: ${styles.hoverBackgroundColor};
+  }
+  &:focus {
+    border: ${styles.focusBorder};
+    background-color: ${styles.focusBackgroundColor};
+  }
+  &:active {
+    border: ${styles.activeBorder};
+    background-color: ${styles.activeBackgroundColor};
+  }
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+  type,
+}) => {
+  const theme = useTheme();
+
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
+    throw new Error(
+      `Error in Button: expected \`children\` of type \`string\` but found ${typeof children}.`,
+    );
+  }
+
+  return (
+    <Flex
+      flex={block ? 1 : 0}
+      flexDirection="row"
+      alignItems="center"
+      alignSelf={styles.align({ align })}
+      justifyContent="center"
+    >
+      <Size height={styles.height({ size, theme })}>
+        <Space padding={styles.padding({ size, children, theme, variant })}>
+          <StyledButton
+            variantColor={variantColor}
+            onClick={onClick}
+            disabled={disabled}
+            size={size}
+            block={block}
+            variant={variant}
+            type={type}
+            {...automation('ds-button')}
+          >
+            <React.Fragment>
+              {icon && iconAlign === 'left' ? (
+                <Icon
+                  name={icon}
+                  size={styles.iconSize({ size, children })}
+                  fill={styles.fontColor({ variant, variantColor, disabled })}
+                />
+              ) : null}
+              {icon && iconAlign === 'left' && children ? (
```

Comment:
Why a separate `View` is conditionally rendered for margin? Can't we do something like this:
```
{icon && iconAlign === 'left' && (
  <Space margin={styles.spaceBetween({ size, iconAlign })}>
    <View>
      <Icon
        name={icon}
        size={styles.iconSize({ size, children })}
        fill={styles.fontColor({ variant, variantColor, disabled })}
      />
    </View>
  </Space>
  )
}
```
Similarly for right icon.

Reply:
True. Missed it.